### PR TITLE
[EBPF-578] gpu: Add systemContext and probe dependencies to system-probe GPU probe

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR refactors the system-probe gpu monitoring module to introduce the `systemContext` struct, which will be used in subsequent PRs to hold system-related information required for the monitoring.

It also uses NVML as a library interface, so that it can be mocked in tests.

### CODE REVIEW

**Review by val06 — CHANGES_REQUESTED** [review]

File: pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe_stub.go [hunk 1]
```diff
@@ -9,10 +9,17 @@
 package ebpfcheck
 
 import (
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
 	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/ebpf/probe/ebpfcheck/model"
 	"github.com/DataDog/datadog-agent/pkg/ebpf"
 )
 
+// ProbeDependencies holds the dependencies for the probe
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe_stub.go]: Mistake while fixing the linter, removed.

File: pkg/gpu/probe.go [hunk 2]
```diff
@@ -36,16 +37,27 @@ const (
...
+	// Telemetry is the telemetry component
+	Telemetry telemetry.Component
+
+	// NvmlLib is the NVML library interface
+	NvmlLib nvml.Interface
+}
+
 // Probe represents the GPU monitoring probe
 type Probe struct {
 	mgr      *ddebpf.Manager
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply pkg/gpu/probe.go:51]: To me the probe "has" an eBPF manager but is not an eBPF manager.

File: cmd/system-probe/modules/gpu_monitoring.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 	"net/http"
```
Comment (val06) [inline cmd/system-probe/modules/gpu_monitoring.go:12]: can we rename that simply go `gpu.go` ? it suits better go naming conventions

Comment (gjulianm) [reply cmd/system-probe/modules/gpu_monitoring.go:12]: Renamed

**Review by val06 — APPROVED** [review]

**Pull request merged**